cfgmgr/intfmgr.cpp

```diff
@@ -843,6 +842,28 @@ bool IntfMgr::doIntfGeneralTask(const vector<string>& keys,
                 m_loopbackIntfList.insert(alias);
                 SWSS_LOG_INFO("Added %s loopback interface", alias.c_str());
             }
+
+            if (adminStatus.empty())
+            {
+                adminStatus = "up";
+            }
+            else if (adminStatus != "up" && adminStatus != "down")
+            {
+                SWSS_LOG_ERROR("Got incorrect value for admin_status as %s for intf %s, defaulting as up", adminStatus.c_str(), alias.c_str());
+                adminStatus = "up";
+            }
+
+            try
+            {
+                string intf_admin = setHostSubIntfAdminStatus(alias, adminStatus, "up");
```

Comment:
why do we call `subintf `api here?

Reply:
the ```subintf```` API functions like a regular set interface admin status API when the parent admin status is passed in as "up", I used this API in order to leverage common code. Changed the approach a bit: modified the code to introduce a new fn setIntfAdminStatus, and called setIntfAdminStatus from within setHostSubIntfAdminStatus to keep common code across both subintf and loopback interface.